app/Http/Controllers/PublicProfileController.php

```diff
@@ -29,6 +29,23 @@ public function index()
             ->with('speakers', $users);
     }
 
+    public function search(Request $request)
+    {
+        $users = User::search($request->search_query)
+            ->orderBy('name', 'asc')->get();
+
+        // Since Scout searches can only perform rudimentary where clauses,
+        // we must filter search results to only validly public profiles.
+        $filteredUsers = $users->filter(function ($user) {
```

Comment:
Huh. So you can't do

```php
$users = User::search($request->input('query'))
    ->where('enable_profile', true)
    ->where('profile_slug', "!=', NULL);
```

.. or something like that? I don't know, but that seems like almost as simple of a where clause as you can get, right?

It's also worth considering whether we only index public profiles, instead of users, which means we'll handle this consideration on the indexing side. Not sure how easy that is.